File: hot100/Solution0001.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
// 给定一个整数数组 nums 和一个整数目标值 target，请你在该数组中找出 和为目标值 target  的那 两个 整数，并返回它们的数组下标。

// 你可以假设每种输入只会对应一个答案，并且你不能使用两次相同的元素。

// 你可以按任意顺序返回答案。

// 方法一：直接嵌套循环
class Solution
{
public:
    std::vector<int> twoSum(const std::vector<int> &nums, int target)
    {
        std::unordered_map<int, int> num_map; // 存储数字及其对应的索引
        for (int i = 0; i < nums.size(); ++i)
        {
            int complement = target - nums[i]; // 计算补数
            if (num_map.find(complement) != num_map.end())
            {
                return {num_map[complement], i}; // 找到目标值，返回索引
            }
            num_map[nums[i]] = i; // 存储当前数字及其索引
        }
        return {}; // 如果没有找到，返回空数组
    }
};

File: hot100/Solution0001.java
package hot100;
// 给定一个整数数组 nums 和一个整数目标值 target，请你在该数组中找出 和为目标值 target  的那 两个 整数，并返回它们的数组下标。

// 你可以假设每种输入只会对应一个答案，并且你不能使用两次相同的元素。

// 你可以按任意顺序返回答案。

import java.util.HashMap;
import java.util.Map;

public class Solution0001 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complete = target - nums[i];
            if (map.containsKey(complete)) {
                return new int[] { map.get(complete), i };
            }
            map.put(nums[i], i);
        }
        return new int[0];
    }

    public static void main(String[] args) {
        Solution0001 solution0001 = new Solution0001();
        int[] nums = { 2, 7, 11, 15 };
        int target = 9;
        int[] result = solution0001.twoSum(nums, target);
        System.out.println("Indices: [" + result[0] + ", " + result[1] + "]");
    }
}
